cmd/asset-syncer/utils.go

```diff
@@ -521,9 +521,12 @@ func pullAndExtract(repoURL *url.URL, appName, tag string, puller helm.ChartPull
 		})
 	}
 
+	// Encode repository names to store them in the database.
+	encodedAppName := url.QueryEscape(appName)
```

Comment:
In other escape/unescape things, we are using Path(Un)Escape instead of this one since the request URI was actually invoked without query params (I mean, the chart name was part of the URI).
AFAIK, QueryEscape performs the escape function on a different subset of characters.

If `appName` comes from a query param, everything is fine. Otherwise, perhaps we can use PathEscape. See what you think.

Reply:
right, the repository name will be part of the URL path so it's better to use PathEscape